lndr-backend/src/Lndr/Db/VerifiedCredits.hs

```diff
@@ -117,7 +121,7 @@ twoPartyBalance addr counterparty ucac conn = do
     return . floor $ balance
 
 
-twoPartyNonce :: Address -> Address -> Connection -> IO Nonce
-twoPartyNonce addr counterparty conn = do
+twoPartyNonce :: Address -> Address -> Integer -> Connection -> IO Nonce
+twoPartyNonce addr counterparty increment conn = do
     [Only nonce] <- query conn "SELECT COALESCE(MAX(nonce) + 1, 0) FROM verified_credits WHERE (creditor = ? AND debtor = ?) OR (creditor = ? AND debtor = ?)" (addr, counterparty, counterparty, addr) :: IO [Only Scientific]
-    return . Nonce . floor $ nonce
+    return ( Nonce ( ( floor $ nonce ) + increment ) )
```

Comment:
You want `return . Nonce . floor $ nonce + increment` here. It's difficult to read with too many parens. Just FYI, the `$` you used here is redundant.

I recommend you practice with haskell's `.` and `$` for a little while. `.` composes functions, `$` is the apply operator which __delays__ evaluation by having very low operator precedence. So I use `$` to separate a bunch of composed functions to a value `x`, like in `f . g . k $ x` (equivalent to `f . g $ k x`), because `.` has a higher operator precedence than `$`. I'll find some tutorial for you that explains this well.